### PR TITLE
Adding, setting and deleting of doc properties

### PR DESCRIPTION
**Pull request by buchi**
Adds functionality to add, set or delete custom document properties.
Document properties are handled in a dict-like way.

```python
document = Document('mydocument.docx')
props = CustomProperties(document)

# Get property value
value = props['My Property']

# Set/add property
props['My Property'] = 'A value'

# Delete property
del props['My Property']
```

### CODE REVIEW

File: docxcompose/properties.py [hunk 1]
```diff
@@ -1,10 +1,65 @@
...
+    elif isinstance(value, string_types):
+        el = parse_xml(CUSTOM_PROPERTY_TYPES['text'])
+        el.text = value
+    else:
+        raise TypeError('Unsupported type {}'.format(type(value)))
+    return el
+
+
+def vt2value(element):
+    tag = element.tag.split('}')[-1]
```
Comment (deiferni) [inline docxcompose/properties.py]: isn't there a more elegant way to get rid of the namespace?

File: docxcompose/properties.py [hunk 2]
```diff
@@ -13,50 +68,144 @@ class CustomProperties(object):
...
         else:
+            self.part = part
             self._element = parse_xml(part.blob)
 
-    def dict(self):
-        """Returns a dict with all custom doc properties"""
-        if self._element is None:
-            return dict()
+    def _part_template(self):
+        template_path = os.path.join(
```
Comment (deiferni) [inline docxcompose/properties.py]: shouldn't you use `setuptools` `pkg_resources` here?

File: docxcompose/properties.py [hunk 3]
```diff
@@ -13,50 +68,144 @@ class CustomProperties(object):
...
+        prop = xpath(
+            self._element,
+            u'.//cp:property[@name="{}"]'.format(key))
+
+        if not prop:
+            raise KeyError(key)
+
+        prop[0].getparent().remove(prop[0])
+        # Renumber pids
+        pid = 2
```
Comment (deiferni) [inline docxcompose/properties.py]: magic constant. why two? i'd use a static variable to describe intent here.

File: docxcompose/properties.py [hunk 4]
```diff
@@ -13,50 +68,144 @@ class CustomProperties(object):
...
+    def __setitem__(self, key, value):
+        """Set the value of a property."""
+        prop = xpath(
+            self._element,
+            u'.//cp:property[@name="{}"]'.format(key))
+        if not prop:
+            self.add(key, value)
+            return
+
+        el = prop[0][0]
```
Comment (deiferni) [inline docxcompose/properties.py]: please do help the reader and assign `prop[0]` to something so we know what it is.

File: docxcompose/properties.py (hunk 1 above)
Comment (buchi) [reply docxcompose/properties.py]: At least I'm not aware of one. Suggestions?

Comment (deiferni) [reply docxcompose/properties.py]: Yes i think his may work: https://lxml.de/api/lxml.etree.QName-class.html
At the cost of an explicit `lxml` dependency which should be already there implicitly as i'm pretty sure it is required by `python-docx`.

File: docxcompose/properties.py (hunk 2 above)
Comment (buchi) [reply docxcompose/properties.py]: Amended.

File: docxcompose/properties.py (hunk 3 above)
Comment (buchi) [reply docxcompose/properties.py]: No idea why. Ask Microsoft 😄 
I've moved it into a static variable.

File: docxcompose/properties.py (hunk 4 above)
Comment (buchi) [reply docxcompose/properties.py]: Changed it to plural.

File: docxcompose/properties.py (hunk 1 above)
Comment (buchi) [reply docxcompose/properties.py]: Using QName now

**Pull request merged**